File: js/src/maplestory/api/common/dto/guild/guildBasic.ts
export abstract class GuildBasicDto {
  public abstract date: Date | null;
  public abstract worldName: string;
  public abstract guildName: string;
  public abstract guildLevel: number;
  public abstract guildFame: number;
  public abstract guildPoint: number;
  public abstract guildMasterName: string;
  public abstract guildMemberCount: number;
  public abstract guildMember: string[];
  public abstract guildSkill: GuildSkillDto[];
  public abstract guildNoblesseSkill: GuildSkillDto[];
}

export abstract class GuildSkillDto {
  public abstract skillName: string;
  public abstract skillDescription: string;
  public abstract skillLevel: number;
  public abstract skillEffect: string;
  public abstract skillIcon: string;
}

File: csharp/MapleStory.OpenAPI/Src/Common/DTO/Guild/GuildBasic.cs
namespace MapleStory.OpenAPI.Common.DTO
{
    public abstract class GuildBasicDTO<TGuildSkill>
    {
        public abstract DateTimeOffset? Date { get; set; }
        public abstract string WorldName { get; set; }
        public abstract string GuildName { get; set; }
        public abstract long GuildLevel { get; set; }
        public abstract long GuildFame { get; set; }
        public abstract long GuildPoint { get; set; }
        public abstract string GuildMasterName { get; set; }
        public abstract long GuildMemberCount { get; set; }
        public abstract List<string> GuildMember { get; set; }
        public abstract List<TGuildSkill> GuildSkill { get; set; }
        public abstract List<TGuildSkill> GuildNoblesseSkill { get; set; }
    }

    public abstract class GuildBasicDTO : GuildBasicDTO<GuildSkillDTO>
    {

    }

    public abstract class GuildSkillDTO
    {
        public abstract string SkillName { get; set; }
        public abstract string SkillDescription { get; set; }
        public abstract long SkillLevel { get; set; }
        public abstract string SkillEffect { get; set; }
        public abstract string SkillIcon { get; set; }
    }
}
